File: Taller Tipos de clases/Go/Clase Abstracta/Empleado.go
abstract class Empleado
{
    public string Id;

    public Empleado()
    {
        Id = "0";
    }

    public Empleado(string id)
    {
        Id = id;
    }

    public Empleado(Empleado e)
    {
        Id = e.Id;
    }

    public abstract double CalcularPago();
}

File: Taller Tipos de clases/JAVA/Clase Abstracta/Empleado.java
abstract class Empleado {

    String id;
    String nombre;

    // Constructor sin parámetros
    public Empleado() {
        id = "01";
        nombre = "Desconocido";
    }

    // Constructor con parámetros
    public Empleado(String id, String nombre) {
        this.id = id;
        this.nombre = nombre;
    }

    // Constructor sobrecargado (solo id)
    public Empleado(String id) {
        this.id = id;
        this.nombre = "Sin nombre";
    }

    // Constructor copia
    public Empleado(Empleado e) {
        this.id = e.id;
        this.nombre = e.nombre;
    }

    // Método abstracto
    abstract double calcularPago();
}
